src/app/settings/shelf-settings/components/edit-shelf-action-panel-configuration/edit-shelf-action-panel-configuration.component.ts

```diff
@@ -51,4 +55,46 @@ export class EditShelfActionPanelConfigurationComponent implements OnInit {
   public cancel(): void {
     this.onCancel.emit()
   }
+
+  public get formValidationTooltip(): string | undefined {
+    const invalidControls = this.findInvalidControls()
```

Comment:
Missing type for `invalidControls`. It is not clear from the line what the type is, so it should be included.
I would think that `findInvalidControls` returns `AbstractControl[]` since that is the type of FormControls, but it returns a `string[]`.

Reply:
Fixed